dotCMS/src/main/java/com/dotmarketing/business/ajax/HostAjax.java

```diff
@@ -24,51 +24,69 @@
 import com.liferay.portal.PortalException;
 import com.liferay.portal.SystemException;
 import com.liferay.portal.model.User;
-
-import org.quartz.SchedulerException;
-
 import java.io.File;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-
 import javax.servlet.http.HttpServletRequest;
+import org.quartz.SchedulerException;
 
 public class HostAjax {
 
 	private HostAPI hostAPI = APILocator.getHostAPI();
 	private UserWebAPI userWebAPI = WebAPILocator.getUserWebAPI();
 
 	public Map<String, Object> findHostsForDataStore(String filter, boolean showArchived, int offset, int count) throws PortalException, SystemException, DotDataException, DotSecurityException {
+		return findHostsForDataStore(filter, showArchived, offset, count, Boolean.FALSE);
+	}
+
+	public Map<String, Object> findHostsForDataStore(String filter, boolean showArchived, int offset, int count,
+													 boolean includeSystemHost) throws PortalException, SystemException, DotDataException, DotSecurityException {
 
-		if(filter.endsWith("*"))
+		if (filter.endsWith("*")) {
 			filter = filter.substring(0, filter.length() - 1);
+		}
 		WebContext ctx = WebContextFactory.get();
 		HttpServletRequest req = ctx.getHttpServletRequest();
-		User user = userWebAPI.getLoggedInUser(req);
+		User user = this.userWebAPI.getLoggedInUser(req);
+
+		List<Host> sites = this.hostAPI.findAll(user, this.userWebAPI.isLoggedToFrontend(req));
+		List<Map<String, Object>> siteResults = new ArrayList<>();
+		Collections.sort(sites, new HostNameComparator());
+
+		Boolean addedSystemSite = false;
+		//Make sure that if we are not using a filter the System Host is on top
+		if ( includeSystemHost && !UtilMethods.isSet(filter) ) {
+			final Host systemSite = this.hostAPI.findSystemHost();
+			if ( APILocator.getPermissionAPI().doesUserHavePermission(systemSite, PermissionAPI.PERMISSION_READ, user) ) {
+				siteResults.add(systemSite.getMap());
+				addedSystemSite = true;
+			}
+		}
 
-		HostAPI hostAPI = APILocator.getHostAPI();
-		List<Host> hosts = hostAPI.findAll(user, userWebAPI.isLoggedToFrontend(req));
-		List<Map<String, Object>> hostResults = new ArrayList<Map<String, Object>>();
-		Collections.sort(hosts, new HostNameComparator());
+		for (Host site : sites) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS135)